### PR TITLE
A hilt makes you not cut yourself

### LINKED ISSUE #429: Migrate to Hilt dependency injection
## Description
Transition to a new Hilt library which is built on top of Dagger specifically for android usage
https://developer.android.com/training/dependency-injection/hilt-android
https://dagger.dev/hilt/

## Dependencies
none

## Definition of done
- [ ] Add Hilt dependency to Gradle
- [ ] Refactor our DI code to use Hilt
- [ ] Use proper scopes

### PR DESCRIPTION
**Pull request by heytherewill**
### Summary
Imagine holding a dagger by the blade. Not nice, eh?

### Technical considerations
Maybe we should review our scoping a bit? I'd like another pair of eyes on that
Also, we need to update our templates :)

### Relationships
Closes #429 

### Pull Request checklist
- [X] The code does not hinder the app's performance
- [X] The code does not introduce regressions
- [X] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] One of our [live or file custom templates](https://github.com/toggl/android/tree/master/.idea/templates) was used in this PR

### Pointers
Go at it commit by commit and pay attention to scoping to see if I have messed something up

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions
Only myself

### CODE REVIEW

**Comment by semanticer** [comment]
The crash is really weird, I couldn't find solution so far. Take a look at https://github.com/chrisbanes/tivi/pull/640/files for inspiration if you haven't already. 

Also, modules can now be objects instead of classes

**Comment by daividssilverio** [comment]
Really liking all the 🔥 🔥 🔥
I'm reading the docs to see about the scoping. Will test the app to see if everything still works alright o/.

File: buildSrc/src/main/java/com/toggl/buildsrc/dependencies.kt [hunk 1]
```diff
@@ -67,6 +67,12 @@ object Libs {
         const val browser = "androidx.browser:browser:1.0.0"
         const val fragment = "androidx.fragment:fragment:1.0.0"
 
+        object Hilt {
+            private const val version = "1.0.0-alpha01"
+            const val viewModel = "androidx.hilt:hilt-lifecycle-viewmodel:$version"
+            const val compiler = "com.google.dagger:hilt-android-compiler:$version"
```
Comment (daividssilverio) [inline buildSrc/src/main/java/com/toggl/buildsrc/dependencies.kt:73]:
```suggestion
            const val compiler = "androidx.hilt:hilt-compiler:$version"
```

Seems to fix the view models injections crashes. IWOMM.

**Comment by daividssilverio** [comment]
One tiny detail, I can't create time entries in this branch of some reason.
I can't navigate to the  `StartEdit` view as well.
The logger tells me I can create time entries by tapping the play/stop button, but they won't appear in the log.

File: buildSrc/src/main/java/com/toggl/buildsrc/dependencies.kt (hunk 1 above)
Comment (heytherewill) [reply buildSrc/src/main/java/com/toggl/buildsrc/dependencies.kt:73]: Copy pasta strikes again

**Pull request merged**